src/Mod/Fem/Gui/Command.cpp

```diff
@@ -1508,6 +1508,121 @@ bool CmdFemCompEmEquations::isActive()
+    // to explicitly set the icon of the used command.
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
+    QList<QAction*> a = pcAction->actions();
+
+    assert(iMsg < a.size());
+    pcAction->setIcon(a[iMsg]->icon());
+}
+
+Gui::Action* CmdFemCompMechEquations::createAction()
+{
+    Gui::ActionGroup* pcAction = new Gui::ActionGroup(this, Gui::getMainWindow());
+    pcAction->setDropDownMenu(true);
+    applyCommandData(this->className(), pcAction);
+
+    QAction* cmd0 = pcAction->addAction(QString());
+    cmd0->setIcon(Gui::BitmapFactory().iconFromTheme("FEM_EquationElasticity"));
+    QAction* cmd1 = pcAction->addAction(QString());
+    cmd1->setIcon(Gui::BitmapFactory().iconFromTheme("FEM_EquationDeformation"));
+
+    _pcAction = pcAction;
+    languageChange();
+
+    pcAction->setIcon(cmd0->icon());
+    int defaultId = 0;
+    pcAction->setProperty("defaultAction", QVariant(defaultId));
+
+    return pcAction;
+}
+
+void CmdFemCompMechEquations::languageChange()
+{
+    Command::languageChange();
+
+    if (!_pcAction)
+        return;
+
+    Gui::CommandManager& rcCmdMgr = Gui::Application::Instance->commandManager();
+
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
+    QList<QAction*> a = pcAction->actions();
+
+    Gui::Command* EquationElasticity = rcCmdMgr.getCommandByName("FEM_EquationElasticity");
+    if (EquationElasticity) {
+        QAction* cmd1 = a[0];
+        cmd1->setText(QApplication::translate("FEM_EquationElasticity",
+                                              EquationElasticity->getMenuText()));
+        cmd1->setToolTip(QApplication::translate("FEM_EquationElasticity",
+                                                 EquationElasticity->getToolTipText()));
+        cmd1->setStatusTip(QApplication::translate("FEM_EquationElasticity",
+                                                   EquationElasticity->getStatusTip()));
+    }
+
+    Gui::Command* EquationDeformation = rcCmdMgr.getCommandByName("FEM_EquationDeformation");
+    if (EquationDeformation) {
+        QAction* cmd0 = a[1];
+        cmd0->setText(
+            QApplication::translate("FEM_EquationDeformation", EquationDeformation->getMenuText()));
+        cmd0->setToolTip(QApplication::translate("FEM_EquationDeformation",
+                                                 EquationDeformation->getToolTipText()));
+        cmd0->setStatusTip(QApplication::translate("FEM_EquationDeformation",
+                                                   EquationDeformation->getStatusTip()));
+    }
+}
+
+bool CmdFemCompMechEquations::isActive()
+{
+    // only if there is an active analysis
+    Fem::FemAnalysis* ActiveAnalysis =
+        FemGui::ActiveAnalysisObserver::instance()->getActiveObject();
+    if (!ActiveAnalysis
+        || !ActiveAnalysis->getTypeId().isDerivedFrom(Fem::FemAnalysis::getClassTypeId()))
+        return false;
```

Comment:
The `ActiveAnalysisObserver` class has the `hasActiveObject` function.

Reply:
Thanks. I could this way also simplify existing code